src/Illuminate/Support/Uri.php

```diff
@@ -0,0 +1,338 @@
+use Illuminate\Support\Traits\Conditionable;
+use Illuminate\Support\Traits\Tappable;
+use League\Uri\Contracts\UriInterface;
+use League\Uri\Uri as LeagueUri;
+use SensitiveParameter;
+use Stringable;
+
+class Uri implements Htmlable, Responsable
+{
+    use Conditionable, Tappable;
+
+    /**
+     * The URL generator resolver.
+     */
+    protected static ?Closure $urlGeneratorResolver = null;
+
+    /**
+     * Create a new parsed URI instance.
+     */
+    public function __construct(UriInterface|Stringable|string $uri = '')
+    {
+        $this->uri = $uri instanceof UriInterface ? $uri : LeagueUri::new((string) $uri);
+    }
+
+    /**
+     * Create a new URI instance.
+     */
+    public static function of(UriInterface|Stringable|string $uri = ''): static
+    {
+        return new static($uri);
+    }
+
+    /**
+     * Get a URI instance of an absolute URL for the given path.
+     */
+    public static function to(string $path): static
+    {
+        return new static(call_user_func(static::$urlGeneratorResolver)->to($path));
+    }
+
+    /**
+     * Get a URI instance for a named route.
+     *
+     * @param  \BackedEnum|string  $name
+     * @param  mixed  $parameters
+     * @param  bool  $absolute
+     * @return static
+     *
+     * @throws \Symfony\Component\Routing\Exception\RouteNotFoundException|\InvalidArgumentException
+     */
+    public static function route($name, $parameters = [], $absolute = true): static
+    {
+        return new static(call_user_func(static::$urlGeneratorResolver)->route($name, $parameters, $absolute));
+    }
+
+    /**
+     * Get the URI's scheme.
+     */
+    public function scheme(): ?string
+    {
+        return $this->uri->getScheme();
+    }
+
+    /**
+     * Get the user from the URI.
+     */
+    public function user(bool $withPassword = false): ?string
+    {
+        if ($withPassword) {
+            return $this->uri->getUserInfo();
+        }
+
+        $userInfo = $this->uri->getUserInfo();
+
+        if (is_null($userInfo)) {
+            return null;
+        }
+
+        return str_contains($userInfo, ':')
+            ? Str::before($userInfo, ':')
+            : $userInfo;
+    }
+
+    /**
+     * Get the password from the URI.
+     */
+    public function password(): ?string
+    {
+        $userInfo = $this->uri->getUserInfo();
+
+        return ! is_null($userInfo) ? Str::after($userInfo, ':') : null;
+    }
+
+    /**
+     * Get the URI's host.
+     */
+    public function host(): ?string
+    {
+        return $this->uri->getHost();
+    }
+
+    /**
+     * Get the URI's port.
+     */
+    public function port(): ?int
+    {
+        return $this->uri->getPort();
+    }
+
+    /**
+     * Get the URI's path.
+     *
+     * Empty or missing paths are returned as a single "/".
+     */
+    public function path(): ?string
+    {
+        $path = trim((string) $this->uri->getPath(), '/');
+
+        return $path === '' ? '/' : $path;
```

Comment:
`Str` to the rescue:
```suggestion
        return Str::of($this->uri->getPath())->trim('/')->replaceMatches('^$', '/');
```

Reply:
@taylorotwell Care to elaborate, why you didn't consider this before merging?